src/modules/commander/Arming/PreFlightCheck/checks/magnetometerCheck.cpp

```diff
@@ -72,13 +74,28 @@ bool PreFlightCheck::magnetometerCheck(orb_advert_t *mavlink_log_pub, vehicle_st
 			}
 		}
 
+		for (uint8_t i = 0; i < ORB_MULTI_MAX_INSTANCES; i++) {
+			uORB::SubscriptionData<estimator_status_s> estimator_status_sub{ORB_ID(estimator_status), i};
+
+			if (estimator_status_sub.get().mag_device_id == static_cast<uint32_t>(device_id)) {
+				if (estimator_status_sub.get().control_mode_flags & (1 << estimator_status_s::CS_MAG_FAULT)) {
```

Comment:
I was going to suggest estimator_status_flags, but I suppose it's not quite worth it with the mag_device_id.

```suggestion
				if (estimator_status_flags_sub.get().cs_mag_fault) {
```